u_roller.go

```diff
@@ -0,0 +1,110 @@
+package tls
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"net"
+	"sync"
+	"time"
+)
+
+type Roller struct {
+	HelloIDs            []ClientHelloID
+	HelloIDMu           sync.Mutex
+	NextHelloID         ClientHelloID
+	TcpDialTimeout      time.Duration
+	TlsHandshakeTimeout time.Duration
+}
+
+// NewRoller creates Roller object with default range of HelloIDs to cycle through until a
+// working/unblocked one is found.
+func NewRoller() (*Roller, error) {
+	tcpDialTimeoutInc, err := getRandUint16()
+	if err != nil {
+		return nil, err
+	}
+	tcpDialTimeoutInc = tcpDialTimeoutInc % 14
+
+	tlsHandshakeTimeoutInc, err := getRandUint16()
+	if err != nil {
+		return nil, err
+	}
+	tlsHandshakeTimeoutInc = tlsHandshakeTimeoutInc % 20
+
+	r := &Roller{
+		HelloIDs: []ClientHelloID{
+			HelloChrome_Auto,
+			HelloFirefox_Auto,
+			HelloIOS_Auto,
+		},
```

Comment:
What about random?  I think it might make sense to include the randomized handshake as the last option, since that's the only one that "guaranteed" to work.

Reply:
Good point.